### DISCUSSION TITLE
Assign role to multiple members in a single API call

### DISCUSSION THREAD

**Discussion by eyagovbusiness** [DSharpPlus/DSharpPlus#1492] [Q&A] [answered]
Hello, I realized after making several API calls to Discord the library is waiting a bit to avoid Disocrd throwing packages to avoid overload from a single source.
Is there a way to assign a role to several members in a single API call? is this supported or will be supported in this library? Having to wait 2 min from assigning roles one by one is not very efficient.

Thank you.

**Comment by VelvetToroyashi** [comment, marked as answer]
Discord's API is, by nature, rate-limited to avoid abuse. Unfortunately no, there is no way to modify multiple members in a single API request. As for "waiting 2 minutes between calls", the library simply follows the [ratelimit headers](https://discord.com/developers/docs/topics/rate-limits#rate-limits) provided by Discord when hitting [this](https://discord.com/developers/docs/resources/guild#modify-guild-member) endpoint.

**Reply by eyagovbusiness** [reply]
Thank you for the answer, do you know if there is a way to check the "X-RateLimit-Limit" from the optional HTTP response headers using DSharpPlus?

**Comment by OoLunar** [comment]
Perhaps you should look into the new linked roles Discord feature? Unfortunately our lib doesn't have full support for it yet (afaik), however it may be of use to you through other resources

**Comment by VelvetToroyashi** [comment]
The library abstracts this away, so no. It'll handle ratelimits for you, so
just make the request and it'll be queued as is appropriate

**Comment by VelvetToroyashi** [comment]
> Perhaps you should look into the new linked roles Discord feature? Unfortunately our lib doesn't have full support for it yet (afaik), however it may be of use to you through other resources

Unfortunately no, and given the HTTP-required nature of it I doubt we'll have true support for it other than entities

**Reply by eyagovbusiness** [reply]
Okay, Thank you again.
